frontend/recipe/native-notification/native-notification.ts

```diff
@@ -0,0 +1,71 @@
+import { css, customElement, html, property } from "lit-element";
+import { nothing } from "lit-html";
+import { Recipe, recipeInfo } from "../recipe";
+import "@vaadin/vaadin-text-field/vaadin-text-field";
+import "@vaadin/vaadin-text-field/vaadin-text-area";
+import "@vaadin/vaadin-button/vaadin-button";
+
+@recipeInfo({
+  url: "native-notification",
+  howDoI: "Show a native OS-level notification",
+  description:
+    "Show how you display a native OS-level notification to the user.",
+})
+@customElement("native-notification")
+export class NativeNotification extends Recipe {
+
+  @property({type: Object})
+  private error?: string;
+
+  render() {
+    return html`
+      <div class="flex-column">
+        <vaadin-text-field label="Title"></vaadin-text-field>
+        <vaadin-text-area label="Body"></vaadin-text-area>
+      </div>
+      <vaadin-button @click=${this.showNotification}>
+        Show Notification
+      </vaadin-button>
+
+      ${this.error ? html`
+        <p>The user has not allowed this page to show notifications:
+        ${this.error}</p>
+      ` : nothing}
+    `;
+  }
+
+  async showNotification() {
+    const title = this.renderRoot.querySelector('vaadin-text-field')!.value;
```

Comment:
IMO the field values should be bound to properties and read from there so you don't reference the element. Now the code would break for instance if the field type changes. We should be mindful of demo code as people will just copy-paste it and build on the same pattern

Reply:
👍